spec/mongoid/extensions/date_spec.rb

```diff
@@ -76,4 +76,22 @@
       expect(date.mongoize).to eq(time)
     end
   end
+
+  describe ".mongoize" do
+    let(:date) do
+      Date.new(2010, 1, 1)
```

Comment:
This file also needs tests for assigning valid and invalid stringified dates & times I think?

The stringified date case can be tricky because there is an implied time zone in which that date would be interpreted.

Reply:
> The stringified date case can be tricky because there is an implied time zone in which that date would be interpreted.

AFAIK Ruby Date class ignores timezone:

```
> Date.parse("2022-01-01")
 => #<Date: 2022-01-01 ((2459581j,0s,0n),+0s,2299161j)>

> Date.parse("2022-01-01Z")
 => #<Date: 2022-01-01 ((2459581j,0s,0n),+0s,2299161j)>

> Date.parse("2022-01-01+0900")
 => #<Date: 2022-01-01 ((2459581j,0s,0n),+0s,2299161j)>
```

Reply:
Will be done in https://jira.mongodb.org/browse/MONGOID-5425.